src/main/java/classify/commands/EditStudent.java

```diff
@@ -218,42 +193,85 @@ private static void openNumberEditInterface(Scanner in, StudentAttributes attrib
      * @param attributes The StudentAttributes object to store the attributes of the
      *                   student.
      */
-    private static void editAttribute(Scanner in, StudentAttributes attributes) {
+    private static void editSubject(Scanner in, StudentAttributes attributes) {
 
-        while (true) {
+        SubjectGrade currentSubject = promptForSubject(in, attributes);
+        updateSubjectName(in, attributes, currentSubject);
+        updateSubjectGrade(in, currentSubject);
+        updateSubjectAttendance(in, currentSubject);
+        //System.out.println("Subject updated.");
+    }
+
+    private static void updateSubjectAttendance(Scanner in, SubjectGrade currentSubject) {
 
+        if (currentSubject == null) {
+            return;
+        }
+        int newClassesAttended = AddStudent.promptForClassesAttended(in);
+        currentSubject.setClassesAttended(newClassesAttended);
+    }
+
+    private static void updateSubjectGrade(Scanner in, SubjectGrade currentSubject) {
+
+        if (currentSubject == null) {
+            return;
+        }
+        double newGrade = InputParsing.promptForGrade(in);
+        if (newGrade != -1) {
+            currentSubject.setGrade(newGrade);
+        }
+    }
+
+    private static SubjectGrade promptForSubject(Scanner in, StudentAttributes attributes) {
+
+        while (true) {
             System.out.print("Subject to edit (enter nothing to exit): ");
             String subjectToFind = in.nextLine().trim();
 
+            //early exit clause
             if (subjectToFind.isBlank()) {
                 System.out.println("No subject edited.");
-                return;
+                return null;
             }
 
             SubjectGrade currentSubject = attributes.findSubject(subjectToFind);
+            
+            if (currentSubject == null) {
+                System.out.println("Subject not found");
+                
+            } else {
+                return currentSubject;
+            }
+        }
+    }
 
-            if (currentSubject != null) {
-                System.out.print("New subject name (enter nothing to skip): ");
-                String newName = in.nextLine().trim();
+    private static void updateSubjectName(Scanner in, StudentAttributes attributes,
+                                   SubjectGrade currentSubject) {
+        if (currentSubject == null) {
+            return;
+        }
+        
+        while (true) {
+            System.out.print("New subject name (enter nothing to skip): ");
+            String newName = in.nextLine().trim();
```

Comment:
perhaps add a check for special characters here? There is a function in InputParsing for that.